main.js

```diff
@@ -0,0 +1,12 @@
+var parser = require('./parser.js').parser,
+    makeXPathModels = require('./models.js').makeXPathModels;
+
+parser.makeXPathModels = makeXPathModels;
+
+parser.setXPathModels = function(models) {
+    parser.yy.xpathmodels = models;
```

Comment:
indeed. I thought the parser was referencing a global `yy`, but I see now it's parameterized. man, that code is convoluted.